string.go

```diff
@@ -248,19 +244,23 @@ func (d *Decoder) getStringLength(tag byte) (int, error) {
 		}
 
 		length = int(b0)<<8 + int(b1)
-		return length, nil
```

Comment:
the length will be valid in these cases, not need to check whether less than zero again below